### PR TITLE
ENH: add RohdeSchwarzPowerSupply

### PR DESCRIPTION
**Pull request by tongju12**
## Description
Add new Rohde SchwarzPowerSupply

## Motivation and Context
https://jira.slac.stanford.edu/browse/LCLSECSD-313
## How Has This Been Tested?
In [1]: from pcdsdevices.rs_powersupply import RohdeSchwarzPowerSupply

In [2]: rs = RohdeSchwarzPowerSupply('TMO:IP1:HMP:01', name='TMO_RSPowersupply')

In [3]: rs.ch2.voltage.get()
Out[3]: 0.0

In [4]: rs.ch1.current.get()
Out[4]: 0.0

## Where Has This Been Documented?
source/upcoming_release_notes/1022-add_RSpowersupply.rst

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on travis
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Comment by ZLLentz** [comment]
Before I start looking at the code- it'd be helpful for documentation purposes to fill in the pull request body. When you created the pull request there were a bunch of title headers and empty sections that are supposed to be filled in. I don't need these to do the review because it's not a big change, but it'd be good for documentation for the future.

I checked some of the boxes in the checklist that I know are complete

I'll link the jira ticket for context, but you should also put this in the "motivation and context" section above:
https://jira.slac.stanford.edu/browse/LCLSECSD-313

**Review by ZLLentz — COMMENTED** [review]
Hi Tong, I left some suggestions. Nothing major needs to be changed but I think a few small tweaks would make it neater. If you end up merging any of my suggestions with the buttons in the github interface make sure you do a `git pull` before editing further.

Good job overall :+1:

I'll mark this as approved and merge once you make further changes or say why you prefer not to change various things.

File: pcdsdevices/rs_powersupply.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    volt_step_rbv = Cpt(EpicsSignalRO, 'VOLT_STEP_RBV', kind='hinted')
+    set_current_rbv = Cpt(EpicsSignalRO, 'SET_CURR_RBV', kind='hinted')
+    set_voltage_rbv = Cpt(EpicsSignalRO, 'SET_VOLT_RBV', kind='hinted')
+    status = Cpt(EpicsSignalRO, 'STATUS', kind='hinted')
+
+
+class RohdeSchwarzPowerSupply(Device):
+    """
+    Class to define PVs of all channels of the RSPowerSupply NGP800.
+    prefix: str
```
Comment (ZLLentz) [inline pcdsdevices/rs_powersupply.py:37]: Nearly there- just a small formatting addition so the auto docs know that these are init parameters and not something else:

```suggestion

    Parameters
    ----------
    prefix: str
```

File: pcdsdevices/rs_powersupply.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+from ophyd import Component as Cpt
+from ophyd import Device, EpicsSignal, EpicsSignalRO
+
+logger = logging.getLogger(__name__)
+
+
+class RSchannel(Device):
+    """
+    Class to define PVs of all channels of the RSPowerSupply NGP800.
+    prefix: str
```
Comment (ZLLentz) [inline pcdsdevices/rs_powersupply.py:12]: Same as the other suggestion:
```suggestion

    Parameters
    ----------
    prefix: str
```

**Review by ZLLentz — COMMENTED** [review]
Last suggestion from me: one small tweak to the docstring so that the auto doc renders nicely

**Comment by ZLLentz** [comment]
Hi Tong, your last commit only targeted one of the two docstrings and had a typo

File: pcdsdevices/rs_powersupply.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import logging
+
+from ophyd import Component as Cpt
+from ophyd import Device, EpicsSignal, EpicsSignalRO
+
+logger = logging.getLogger(__name__)
+
+
+class RSchannel(Device):
```
Comment (ZryletTC) [inline pcdsdevices/rs_powersupply.py:9]: I believe this should be capitalized, right? If so, make sure to change the other instances as well.
```suggestion
class RSChannel(Device):
```

File: pcdsdevices/rs_powersupply.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    """
+    Class to define PVs of all channels of the RSPowerSupply NGP800.
```
Comment (ZryletTC) [inline pcdsdevices/rs_powersupply.py:11]:
```suggestion
    Class to define PVs for a single channel of the Rohde-Schwarz NGP800.
```

File: pcdsdevices/rs_powersupply.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    prefix: str
+        The base PV of the relevant RohdeSchwarzPowersupply.
+    name: str, keyword-only
+        A name to refer to the relevant RohdeSchwarzPowersupply
```
Comment (ZryletTC) [inline pcdsdevices/rs_powersupply.py:15]: formatting and channel vs powersupply
```suggestion
    prefix : str
        The base PV of the relevant Rohde-Schwarz channel.
    name : str, keyword-only
        A name to refer to the relevant Rohde-Schwarz channel.
```

File: pcdsdevices/rs_powersupply.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+class RohdeSchwarzPowerSupply(Device):
+    """
+    Class to define PVs of all channels of the RSPowerSupply NGP800.
+    Paramters
+    ----------
+    prefix: str
+        The base PV of the relevant RohdeSchwarzPowersupply.
+    name: str, keyword-only
+        A name to refer to the relevant RohdeSchwarzPowersupply
```
Comment (ZryletTC) [inline pcdsdevices/rs_powersupply.py:42]: same formatting changes as above
```suggestion
    prefix : str
        The base PV of the relevant RohdeSchwarzPowersupply.
    name : str, keyword-only
        A name to refer to the relevant RohdeSchwarzPowersupply.
```

**Comment by ZLLentz** [comment]
Hi @tongju12 , do you have time to implement Tyler's suggestions? Or should I take care of this last cleanup step when I have a chance?

File: pcdsdevices/rs_powersupply.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+from ophyd import Device, EpicsSignal, EpicsSignalRO
+
+logger = logging.getLogger(__name__)
+
+
+class RSChannel(Device):
+    """
+    Class to define PVs for a single channel of the RSPowerSupply NGP800.
+
+    Paramters
```
Comment (ZLLentz) [inline pcdsdevices/rs_powersupply.py:13]:
```suggestion
    Parameters
```

File: pcdsdevices/rs_powersupply.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    volt_step_rbv = Cpt(EpicsSignalRO, 'VOLT_STEP_RBV', kind='hinted')
+    set_current_rbv = Cpt(EpicsSignalRO, 'SET_CURR_RBV', kind='hinted')
+    set_voltage_rbv = Cpt(EpicsSignalRO, 'SET_VOLT_RBV', kind='hinted')
+    status = Cpt(EpicsSignalRO, 'STATUS', kind='hinted')
+
+
+class RohdeSchwarzPowerSupply(Device):
+    """
+    Class to define PVs of all channels of the RSPowerSupply NGP800.
+    Paramters
```
Comment (ZLLentz) [inline pcdsdevices/rs_powersupply.py:46]:
```suggestion

    Parameters
```

**Review by ZLLentz — APPROVED** [review]
Should have merged earlier, merging now because these commits are embedded into #1058 and I want to keep things organized

**Pull request merged**